docs/decisions/broker-trigger.md

```diff
@@ -0,0 +1,137 @@
+# Broker and Trigger model (for 0.5.0)
+
+Decision date: 6 March 2019
+
+## Define High-Level Objects Related to Eventing
+
+The goal of eventing is to provide an abstract "bucket of events" which event
+consumers can sample from using filters on CloudEvent
+[context attributes](https://github.com/cloudevents/spec/blob/master/spec.md#context-attributes).
+Event producers can deliver events into the bucket without needing to understand
+the exact mechanics of routing to specific consumers. Event producers and
+consumers are decoupled temporally but also through a deliberately black-box
+abstraction which can optimize the underlying routing layer dynamically.
+
+## Objects
+
+The following objects are the minimum set needed to experiment and collect
+customer feedback on the Eventing object model, aka an MVP (including rough YAML
+definitions; final naming and field definitions subject to change based on
+implementation experience).
+
+### Broker
+
+A Broker represents a "bucket of events". It is a namespaced object, with
+automation provided to provision a Broker named `default` in
+appropriately-labelled namespaces. It is expected that most users will not need
+to explicitly create a Broker and simply enabling eventing in the namespace to
+create the `default` Broker will be sufficient. Assuming that the Broker model
+is successful, future recommendations for Source CRD authors would be to default
+the Source's `spec.sink` attribute to point to the `default` Broker in the
+namespace.
+
+Historical storage and replay of events is not part of the Broker design or MVP.
+
+Broker implements the "Addressable" duck type to allow Source custom resources
+to target the Broker. Broker DNS names should also be predictable (e.g.
+`default-broker` for the `default` Broker), to enable whitebox event producers
+to target the Broker without needing to reconcile against the "Addressable" duck
+type.
+
+#### Broker Object definition
+
+```golang
+type Broker struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec BrokerSpec `json:"spec,omitempty"`
+	Status BrokerStatus `json:"status,omitempty"`
+}
+
+type BrokerSpec struct {
+  // Empty for now
+}
+
+type BrokerStatus struct {
+	Conditions duckv1alpha1.Conditions `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	Address duckv1alpha1.Addressable `json:"address,omitempty"`
+}
+```
+
+### Trigger
+
+A Trigger represents a registration of interest (request for delivery to an
+Addressable object or URL) in a filtered selection of events delivered to a
+Broker. For the MVP, Triggers may only target Addressable objects in the same
+namespace.
+
+#### Buffering
+
+Triggers are assumed to buffer (durably queue) filtered events from the Broker
```

Comment:
Should we omit Buffering sub section or include it as a general requirement rather than Trigger specific?
I read the rest of the document as describing an API model. The Broker as a pool/bucket of events, while triggers define an intent to deliver filtered events to addressable objects.
"Buffering" seems to be part of the implementation detail of the routing layer.
So either omit it, or call it out as a general requirement of the routing layer rather than call it as a sub section of Trigger.

Thoughts?

Reply:
See the example I posted here:

https://github.com/knative/eventing/pull/862#discussion_r264804835

TL;DR: Event buffering is independent for each Trigger; Triggers do not see events from before they are created, even if some other Trigger has not yet delivered the event and still has it buffered.

I think this is an important property which it is worthwhile to document.